test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java

```diff
@@ -71,4 +73,12 @@ public void beforeEach(QuarkusTestMethodContext context) {
         }
 
     }
+
+    private Principal createPrincipal(TestSecurity testSecurity) {
+        Instance<PrincipalBuilder> producer = CDI.current().select(PrincipalBuilder.class);
+        if (producer.isResolvable()) {
```

Comment:
@mkouba Hi Martin - here, in the debug mode, I see the `Instance<PrincipalBuilder> producer` containing a resolved `JwtPrincipalBuilderProducer` but `producer.isResolvable()` is failing - can you see what I'm missing :-) ?

Reply:
I think that the dependency is ambiguous because there are two beans that match the `PrincipalBuilder` type: 1. the `JwtPrincipalBuilderProducer` class bean, 2.  the `JwtPrincipalBuilderProducer.produce()` producer bean. Bean 1 is not removed because it declares a producer method that is unremovable...

Reply:
@mkouba Oh I see, thanks Martin :-), should be easy to fix